### PR TITLE
[core] Remove use of getState

### PR DESCRIPTION
**Pull request by oliviertassinari**
A follow-up on https://github.com/mui-org/material-ui-x/pull/2208#discussion_r680924729. I don't get why we even have such `getState` method. For #820, I think that we could restructure the state, and maybe have `getState` returns a subset of the state that is public.

### CODE REVIEW

**Comment by flaviendelangle** [comment]
@m2mathew do you remember exactly what was the purpose of `.getState()` and if it makes sense to use it in our codebase ?
I agree that it's lighter to just access the property, but I may be missing something here.

**Comment by oliviertassinari** [comment]
We had a getState(subpart: string) API at some point 🤷‍♂️, happy with whatever direction we take here.

**Review by m4theushw — APPROVED** [review]
We already removed in #2141 the ability to return a subset of the state because it was never used. I'm OK with removing this method or making it private.

File: packages/grid/_modules_/grid/models/api/gridStateApi.ts [hunk 1]
```diff
@@ -8,8 +8,7 @@ export interface GridStateApi {
   state: GridState;
   /**
    * Returns the state of the grid.
-   * @param {string} stateId The part of the state to be returned.
-   * @returns {any} The state of the grid.
+   * @returns {GridState} The state of the grid.
```
Comment (m4theushw) [inline packages/grid/_modules_/grid/models/api/gridStateApi.ts:11]: Should we not document it for now?

```suggestion
   * @returns {GridState} The state of the grid.
   * @ignore - do not document.
```

**Comment by flaviendelangle** [comment]
If we stop using it internally, we can just remove it.

File: packages/grid/_modules_/grid/models/api/gridStateApi.ts (hunk 1 above)
Comment (flaviendelangle) [reply packages/grid/_modules_/grid/models/api/gridStateApi.ts:11]: If we stop using it internally, we can just remove it.

**Comment by m4theushw** [comment]
@oliviertassinari I've resolved the conflicts, removed a few remaining `getState` and made it private.

File: packages/grid/_modules_/grid/models/api/gridStateApi.ts (hunk 1 above)
Comment (oliviertassinari) [reply packages/grid/_modules_/grid/models/api/gridStateApi.ts:11]: @flaviendelangle Removing the method would be a breaking change. We can move more iteratively.

**Comment by oliviertassinari** [comment]
@m4theushw Thanks

**Pull request merged**